website/client/components/challenges/challengeModal.vue

```diff
@@ -285,26 +295,36 @@ export default {
       };
     },
     async createChallenge () {
-      if (!this.workingChallenge.name) alert('Name is required');
-      if (!this.workingChallenge.description) alert('Description is required');
-
-      this.workingChallenge.timestamp = new Date().getTime();
+      // @TODO: improve error handling, add it to updateChallenge, make errors translatable. Suggestion: `<% fieldName %> is required` where possible, where `fieldName` is inserted as the translatable string that's used for the field header.
+      let errors = '';
+      if (!this.workingChallenge.name) errors += 'Name is required\n';
```

Comment:
No, this whole error handling section needs to be improved, as mentioned in that TODO comment so it would be best to create the translatable strings at the same time that the final error handling code is added (the strings currently here now aren't likely to stay). It's the same problem that exists in the group creation vue file, so ideally the same method of error handling would be added for both.